src/DataCollector/ProxyFactory.php

```diff
@@ -0,0 +1,242 @@
+<?php
+
+/*
+ * This file is part of php-cache\cache-bundle package.
+ *
+ * (c) 2015-2015 Aaron Scherer <aequasi@gmail.com>, Tobias Nyholm <tobias.nyholm@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Cache\CacheBundle\DataCollector;
+
+/**
+ * Generate proxies over your cache pool. This should only be used in development.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class ProxyFactory
+{
+    /**
+     * @type string
+     */
+    private $proxyDirectory;
+
+    /**
+     * @param string $proxyDirectory
+     */
+    public function __construct($proxyDirectory)
+    {
+        $this->proxyDirectory = $proxyDirectory;
+    }
+
+    /**
+     * Create a proxy that handles logging better.
+     *
+     * @param string $class
+     * @param string $file  where we store the proxy class
+     *
+     * @return string the name of a much much better class
+     */
+    public function createProxy($class, &$proxyFile = null)
+    {
+        $proxyClass = $this->getProxyClass($class);
+        $class      = '\\'.rtrim($class, '\\');
+        $proxyFile  = $this->proxyDirectory.'/'.$proxyClass.'.php';
+
+        if (class_exists($proxyClass)) {
+            return $proxyClass;
+        }
+
+        $content = <<<PROXY
```

Comment:
Could we `get_file_contents()` this file instead? It will be easier to change and review.